File: src/main/java/org/example/expert/domain/log/service/LogService.java
package org.example.expert.domain.log.service;


import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.example.expert.domain.log.entity.Log;
import org.example.expert.domain.log.repository.LogRepository;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Propagation;
import org.springframework.transaction.annotation.Transactional;

@Service
@RequiredArgsConstructor
@Slf4j
public class LogService {
    private final LogRepository logRepository;

    @Transactional(propagation = Propagation.REQUIRES_NEW)
    public void saveLog(String message) {
        log.info("saving log : {}", message);
        logRepository.save(new Log(message));
    }
}

File: src/main/java/org/example/expert/domain/log/service/LogService.kt
package org.example.expert.domain.log.service

import org.example.expert.domain.log.entity.Log
import org.example.expert.domain.log.repository.LogRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Propagation
import org.springframework.transaction.annotation.Transactional


@Service
class LogService(
    private val logRepository: LogRepository
) {
    @Transactional(propagation = Propagation.REQUIRES_NEW)
    fun saveLog(message: String) {
        val log = Log(message)
        logRepository.save(log)
    }
}
